pkgs/development/libraries/blkhash/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, cmake
+, openssl
+, libnbd
+, asciidoc
+, python3Packages
+}:
+let
+  unity = fetchFromGitHub {
+    owner = "ThrowTheSwitch";
+    repo = "Unity";
+    rev = "5204c1bacf37e0c38211d2fbb6d9796e410223f8";
+    sha256 = "i4uQOG/Q86AaQINJoC/65QA/ZEUOXzaVceR2inw32yU=";
+  };
+in
+stdenv.mkDerivation rec {
+  version = "0.6.1";
+  pname = "blkhash";
+
+  src = fetchFromGitLab {
+    owner = "nirs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "+7SIZdvkdMYVgeklbqU0MHKc1VJnk9opqhsCmkg6bcI=";
+  };
```

Comment:
Does this fetch the code from gitlab or download a tarball?

Reply:
It downloads the tarball with the source code :)